src/libraries/System.Runtime/tests/System.Runtime.Tests/System/StringTests.cs

```diff
@@ -1347,36 +1347,57 @@ public static void ImplicitCast_NullString_ReturnsDefaultSpan()
         }
 
         [Theory]
-        [InlineData("Hello", 'l', StringComparison.Ordinal, 2)]
-        [InlineData("Hello", 'x', StringComparison.Ordinal, -1)]
-        [InlineData("Hello", 'h', StringComparison.Ordinal, -1)]
-        [InlineData("Hello", 'o', StringComparison.Ordinal, 4)]
-        [InlineData("Hello", 'h', StringComparison.OrdinalIgnoreCase, 0)]
-        [InlineData("HelLo", 'L', StringComparison.OrdinalIgnoreCase, 2)]
-        [InlineData("HelLo", 'L', StringComparison.Ordinal, 3)]
-        [InlineData("HelLo", '\0', StringComparison.Ordinal, -1)]
-        [InlineData("!@#$%", '%', StringComparison.Ordinal, 4)]
-        [InlineData("!@#$", '!', StringComparison.Ordinal, 0)]
-        [InlineData("!@#$", '@', StringComparison.Ordinal, 1)]
-        [InlineData("!@#$%", '%', StringComparison.OrdinalIgnoreCase, 4)]
-        [InlineData("!@#$", '!', StringComparison.OrdinalIgnoreCase, 0)]
-        [InlineData("!@#$", '@', StringComparison.OrdinalIgnoreCase, 1)]
-        [InlineData("_____________\u807f", '\u007f', StringComparison.Ordinal, -1)]
-        [InlineData("_____________\u807f__", '\u007f', StringComparison.Ordinal, -1)]
-        [InlineData("_____________\u807f\u007f_", '\u007f', StringComparison.Ordinal, 14)]
-        [InlineData("__\u807f_______________", '\u007f', StringComparison.Ordinal, -1)]
-        [InlineData("__\u807f___\u007f___________", '\u007f', StringComparison.Ordinal, 6)]
-        [InlineData("_____________\u807f", '\u007f', StringComparison.OrdinalIgnoreCase, -1)]
-        [InlineData("_____________\u807f__", '\u007f', StringComparison.OrdinalIgnoreCase, -1)]
-        [InlineData("_____________\u807f\u007f_", '\u007f', StringComparison.OrdinalIgnoreCase, 14)]
-        [InlineData("__\u807f_______________", '\u007f', StringComparison.OrdinalIgnoreCase, -1)]
-        [InlineData("__\u807f___\u007f___________", '\u007f', StringComparison.OrdinalIgnoreCase, 6)]
-        public static void IndexOf_SingleLetter_StringComparison(string s, char target, StringComparison stringComparison, int expected)
-        {
-            Assert.Equal(expected, s.IndexOf(target, stringComparison));
-            var charArray = new char[1];
-            charArray[0] = target;
-            Assert.Equal(expected, s.AsSpan().IndexOf(charArray, stringComparison));
+        [InlineData("Hello", 'l', 0, int.MaxValue, StringComparison.Ordinal, 2)]
```

Comment:
We need to add test cases for empty string with different string comparison options. and ensure the results is correct and the exception thrown when expected.